components/automate-cli/cmd/chef-automate/start_linx_test.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/chef/automate/components/automate-cli/pkg/status"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStartForFrontEndNodes(t *testing.T) {
+	testCases := []struct {
+		args          []string
+		sshUtil       SSHUtil
+		frontendIps   []string
+		remoteService string
+		isError       bool
+		err           error
+	}{
+		{
+			[]string{"some_args"},
+			getMockSSHUtil(&SSHConfig{}, nil, "", nil),
+			[]string{"127.0.1.3", "127.0.1.4", "127.0.1.5"},
+			"automate",
+			false,
+			nil,
+		},
+		{
+			[]string{"some_args"},
+			getMockSSHUtil(&SSHConfig{}, nil, "", nil),
+			argsEmpty,
+			"chef-server",
+			true,
+			status.Errorf(1, "No chef-server IPs are found"),
+		},
+		{
+			[]string{"some_args"},
+			NewSSHUtil(&SSHConfig{}),
+			argsEmpty,
+			"automate",
+			true,
+			errors.Errorf("No automate IPs are found"),
+		},
+		{
+			[]string{"some_args"},
+			getMockSSHUtil(&SSHConfig{}, nil, "", errors.New("Process exited with status 1")),
+			[]string{"127.1.0.3"},
+			"chef-server",
+			true,
+			errors.New("Not able to start one or more nodes in chef-server: \nopen : no such file or directory"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := checkNodes(testCase.args, testCase.sshUtil, testCase.frontendIps, testCase.remoteService, getMockWriterImpl())
+		if testCase.isError {
+			assert.EqualError(t, testCase.err, err.Error())
+		}
+	}
+}
+
+func TestStartForBackEndNodes(t *testing.T) {
+	testCases := []struct {
+		args          []string
+		sshUtil       SSHUtil
+		frontendIps   []string
+		remoteService string
+		isError       bool
+		err           error
+	}{
+		{
+			[]string{"some_args"},
+			getMockSSHUtil(&SSHConfig{}, nil, "", nil),
+			[]string{"127.0.0.3", "127.0.0.4", "127.0.0.5"},
+			"opensearch",
+			false,
+			nil,
+		},
+		{
+			[]string{"some_args"},
+			getMockSSHUtil(&SSHConfig{}, nil, "", nil),
+			[]string{},
+			"postgresql",
+			true,
+			status.Errorf(1, "No postgresql IPs are found"),
+		},
+		{
+			[]string{"some_args"},
+			getMockSSHUtil(&SSHConfig{}, nil, "", errors.New("Process exited with status 1")),
+			[]string{"127.0.1.3"},
+			"postgresql",
+			true,
+			errors.New("Not able to start one or more nodes in postgresql: \nopen : no such file or directory"),
```

Comment:
This error should not be different from what you have passed in MockSShUtil
MockSSHUtil: Process exited with status 1
TestExpectedResult: open : no such file or directory